partcad/src/partcad/ai_feature_file.py

```
@@ -0,0 +1,92 @@
+        self.is_image = False
+        self.is_video = False
+        self.is_pdf = False
+        self.suffix = ".txt"
+        self.filename = None
+
+        # Switch to lowercase to normalize the file extension in the URL
+        lower_url = url.lower()
+        if lower_url.startswith(("http://", "https://")):
+            response = requests.get(self.url, allow_redirects=True, timeout=15)
+            if response.status_code == 200:
+                self.filename = tempfile.NamedTemporaryFile(suffix=self.suffix, delete=True, delete_on_close=False)
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)